### PR TITLE
v1.0.1

### PR DESCRIPTION
**Pull request by alexey-troshkov**

### CODE REVIEW

File: android/src/main/kotlin/io/piano/flutter/piano_analytics/PianoAnalyticsPlugin.kt [hunk 1]
```diff
@@ -68,24 +66,47 @@ class PianoAnalyticsPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
...
     }
 
     override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
         channel.setMethodCallHandler(null)
-        consentsPlugin.onAttachedToEngine(binding)
     }
 
     override fun onMethodCall(call: MethodCall, result: Result) {
         try {
+            var skipResult = false
```
Comment (DeKaN) [inline android/src/main/kotlin/io/piano/flutter/piano_analytics/PianoAnalyticsPlugin.kt:77]:
```suggestion
            val skipResult = call.method == "getVisitorId"
```

File: android/src/main/kotlin/io/piano/flutter/piano_analytics/PianoAnalyticsPlugin.kt [hunk 2]
```diff
@@ -97,10 +118,19 @@ class PianoAnalyticsPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
...
+                visitorStorageLifetime = call.argument<Int>("storageLifetimeVisitor")
+                    ?: Configuration.DEFAULT_VISITOR_STORAGE_LIFETIME,
+                visitorStorageMode = getVisitorStorageMode(call.argument("visitorStorageMode")),
+                ignoreLimitedAdTracking = call.argument<Boolean>("ignoreLimitedAdvertisingTracking")
+                    ?: false
             ).build(),
             pianoConsents = getPianoConsents()
         )
+        val visitorId = call.argument<String>("visitorId")
+        if (visitorId != null) {
```
Comment (DeKaN) [inline android/src/main/kotlin/io/piano/flutter/piano_analytics/PianoAnalyticsPlugin.kt:131]: There should be a check for visitorIDType == custom (or forcing custom type)

File: android/src/main/kotlin/io/piano/flutter/piano_analytics/PianoAnalyticsPlugin.kt [hunk 3]
```diff
@@ -109,16 +139,92 @@ class PianoAnalyticsPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
...
+        PianoAnalytics.getInstance().userStorage.currentUser = User(
+            call.arg("id"),
+            category = call.argument("category"),
+            shouldBeStored = call.argument<Boolean?>("enableStorage") ?: true
+        )
+    }
+
+    private fun handleGetVisitorId(result: Result) {
+        val pianoAnalytics = PianoAnalytics.getInstance()
+        result.success(pianoAnalytics.customVisitorId ?: pianoAnalytics.visitorId)
```
Comment (DeKaN) [inline android/src/main/kotlin/io/piano/flutter/piano_analytics/PianoAnalyticsPlugin.kt:186]: There should be a check for visitorIdType not for null value in customVisitorId

File: android/src/main/kotlin/io/piano/flutter/piano_analytics/PianoAnalyticsPlugin.kt (hunk 1 above)
Comment (alexey-troshkov) [reply android/src/main/kotlin/io/piano/flutter/piano_analytics/PianoAnalyticsPlugin.kt:77]: skipResult is also used for getUser

File: android/src/main/kotlin/io/piano/flutter/piano_analytics/PianoAnalyticsPlugin.kt (hunk 2 above)
Comment (alexey-troshkov) [reply android/src/main/kotlin/io/piano/flutter/piano_analytics/PianoAnalyticsPlugin.kt]: added check

File: android/src/main/kotlin/io/piano/flutter/piano_analytics/PianoAnalyticsPlugin.kt (hunk 3 above)
Comment (alexey-troshkov) [reply android/src/main/kotlin/io/piano/flutter/piano_analytics/PianoAnalyticsPlugin.kt]: added check

File: android/src/main/kotlin/io/piano/flutter/piano_analytics/PianoAnalyticsPlugin.kt (hunk 1 above)
Comment (DeKaN) [reply android/src/main/kotlin/io/piano/flutter/piano_analytics/PianoAnalyticsPlugin.kt:78]: `val skipResult = call.method in METHODS_WITH_CUSTOM_RESULT`, but it's better to make something like:
```
val methodResult = when (call.method) { ... }
result.success(methodResult)
```
+
```
    private fun handleGetUser(): Map<String?, Any>? {
        return PianoAnalytics.getInstance().userStorage.currentUser?.let {
            mapOf(
                    "id" to it.id,
                    "category" to it.category
            )
        }
    }
```
+
```
private fun handleInit(call: MethodCall): Any? {
...
    return null
}

**Pull request merged**